src/__tests__/type.js

```diff
@@ -1066,3 +1066,274 @@ test('navigation key: {arrowleft} and {arrowright} moves the cursor', () => {
     input[value="abc"] - keyup: c (99)
   `)
 })
+
+test('can type into an input with type `time`', () => {
+  const {element, getEventSnapshot} = setup('<input type="time" />')
+  userEvent.type(element, '0105')
+  expect(getEventSnapshot()).toMatchInlineSnapshot(`
+  Events fired on: input[value="01:05"]
+  
+  input[value=""] - pointerover
+  input[value=""] - pointerenter
+  input[value=""] - mouseover: Left (0)
+  input[value=""] - mouseenter: Left (0)
+  input[value=""] - pointermove
+  input[value=""] - mousemove: Left (0)
+  input[value=""] - pointerdown
+  input[value=""] - mousedown: Left (0)
+  input[value=""] - focus
+  input[value=""] - focusin
+  input[value=""] - pointerup
+  input[value=""] - mouseup: Left (0)
+  input[value=""] - click: Left (0)
+  input[value=""] - keydown: 0 (48)
+  input[value=""] - keypress: 0 (48)
+  input[value=""] - keyup: 0 (48)
+  input[value=""] - keydown: 1 (49)
+  input[value=""] - keypress: 1 (49)
+  input[value=""] - keyup: 1 (49)
+  input[value=""] - keydown: 0 (48)
+  input[value=""] - keypress: 0 (48)
+  input[value="01:00"] - input
+    "{CURSOR}" -> "{CURSOR}01:00"
+  input[value="01:00"] - change
+  input[value="01:00"] - keyup: 0 (48)
+  input[value="01:00"] - keydown: 5 (53)
+  input[value="01:00"] - keypress: 5 (53)
+  input[value="01:05"] - input
+    "{CURSOR}01:00" -> "{CURSOR}01:05"
+  input[value="01:05"] - change
+  input[value="01:05"] - keyup: 5 (53)
+  `)
+  expect(element.value).toBe('01:05')
+})
+
+test('can type more a number higher than 60 minutes into an input `time` and they are converted into 59 minutes', () => {
+  const {element, getEventSnapshot} = setup('<input type="time" />')
+  userEvent.type(element, '2390')
```

Comment:
If I type `2390` in the `time` input here: https://codesandbox.io/s/user-event-playground-eo909?file=/index.html

That doesn't end up firing a change event because I need to type in `a` (for AM) or `p` (for PM). It's possible that this is a user-agent thing, in which case I think the implementation here is fine. In any case, we'll want to make sure we document how a time is intended to be entered for a given time value.

I'm thinking that rather than having them provide `0105` we should have them pass the value they want the `input.value` to be when the typing is complete. So that would be: `userEvent.type(element, '01:05')`. I think that would be more intuitive. We'd just need to make sure we handle this internally because the end user wouldn't actually bother typing the `:` I think.

Reply:
> That doesn't end up firing a change event because I need to type in a (for AM) or p (for PM). It's possible that this is a user-agent thing, in which case I think the implementation here is fine. In any case, we'll want to make sure we document how a time is intended to be entered for a given time value.

In my case trying with chrome I don't have to type `a` or `p`. I guess you're right, it depends on the browser or user-agent, or something else.

> I'm thinking that rather than having them provide 0105 we should have them pass the value they want the input.value to be when the typing is complete. So that would be: userEvent.type(element, '01:05'). I think that would be more intuitive. We'd just need to make sure we handle this internally because the end user wouldn't actually bother typing the : I think.

_technically_ speaking, all the non-digit characters are ignored, as that's what happens when using the `<input type="time" />`. So I could rewrite the tests to use `:` and they will work with the current implementation (and the docs could be with that). What do you think?

if that's enough as the request, I could rewrite the tests and the docs (yet to be added) to show the examples using  `:`. That's probably better in terms of DX, but as you say the user would never type `:`. Working without `:` is an extra in that sense

If you think it is better to change the implementation to just split hours and minutes by `:`, then I can do it; in this scenario, the user would _always_ have to type `:` or it wouldn't work (with the current approach, both works). and I'd have to validate both separatedly

(I also went without `:` probably biased by the example in the ticket, which was not using it).

Reply:
That all sounds great. Thank you. I think having the `:` in the docs would be preferable/sensible. And having at least one test that uses `:` as well as at least one that does not would be good.

Thanks for all of this work!

Reply:
changes applied in https://github.com/testing-library/user-event/pull/502/commits/8333075b4da7bebb856904a748f0c87992d82940

- updated docs to have an example of using `<input type="time />`. My main doubt is that there are no roles associated to this type of input, so the only way to query it was with a `data-testid`. Should I use another option?

 - I updated all tests to use `01:05`, and also added one without the `:` to verify it works in both scenarios

Reply:
to clarify, I tried with labels in testing-playground but it did not work either - that's why I went for `data-testid`

Reply:
How about this? https://testing-playground.com/gist/7b5d922642a0a1814209c81db6867b59/1b0329837c8afa1d5e972e219ed65ba041f69c82

Reply:
I swear I tried it and I was getting an error 🤔  but 🤷 
updated in https://github.com/testing-library/user-event/pull/502/commits/774a0303a7ca3b9e6a0fa3e08c701febc8f7db99